polaris-react/src/components/Banner/Banner.tsx

```diff
@@ -36,6 +36,8 @@ export interface BannerProps {
   title?: string;
   /** Icon to display in the banner. Use only major, duotone icons */
   icon?: IconProps['source'];
+  /** Renders the banner without an icon. */
+  hideIcon?: boolean;
```

Comment:
I was a little bit confused which icon is being hidden. Is it the close icon or the status icon? Maybe some more specificity here could remove that confusion.

Reply:
should we then change the `icon` prop to `statusIcon` as well? or just leave that one since it would be a breaking change?

Reply:
Oof. I see. That connection makes sense. Lets leave it as is. Maybe we can add a slight clarification to the comment above the props.